### ISSUE TITLE
How to parse a vector of objects with const attributes

### ISSUE THREAD

**Issue by axx** [nlohmann/json#1453]
Hi,

I have this class:
```cpp
class MyClass {
  public:
  const std::string _a;
  const std::string _b;
}
```
I then want to parse `[{"a":"aaaaa", "b":"bbbbb"}]` to a `std::vector<MyClass>`.
I tried using the to_json/from_json approach in but I can't because
```cpp
void from_json(const json& j, MyClass& m) {
  m._a = j.at("a");   // error
 m._b = j.at("b");
}
```
The other approach using
```cpp
MyClass from_json(const json& j) {
   return { j.at("a").get<std::string>(), j.at("b").get<std::string>() }
}
```
also doesn't work because of some requirements in std::vector.

Does someone have other ideas aside from just dropping the const in MyClass?

**Comment by theodelrieu** [comment]
Hello,

You technically *could* workaround that by specializing `nlohmann::adl_serializer` for `MyClass`.
Although it would not solve the `std::vector` problem (unless specializing `nlohmann::adl_serializer<std::vector<MyClass>>`, which is quite brittle especially if you want to support other containers).

I know that everytime I tried to use `const` member variables, I ended up removing `const`.
Since it prevents `MoveAssignable` and `CopyAssignable`, lots of containers will not work with it.

Removing `const` and using `MyClass const&` when needed seems better.
